src/routes/jira/workspaces/repositories/jira-workspaces-repositories-get.ts

```diff
@@ -1,3 +1,85 @@
-export const JiraWorkspacesRepositoriesGet = () => {
-	// Content included in ARC-2116-search-connected-repos
+import { Request, Response } from "express";
+import { Errors } from "config/errors";
+import { Subscription } from "models/subscription";
+import { RepoSyncState } from "models/reposyncstate";
+
+export interface WorkspaceRepo {
+	id: string,
+	name: string,
+	workspaceId: string
+}
+
+const DEFAULT_PAGE_NUMBER = 1; // Current page
+export const DEFAULT_LIMIT = 20; // Number of items per page
+
+const getReposForWorkspaceId = async (
+	jiraHost: string,
+	connectedOrgId: number,
+	page: number,
+	limit: number,
+	repoName?: string
+): Promise<RepoSyncState[] | null> => {
+	const subscription = await Subscription.getOneForSubscriptionIdAndHost(jiraHost, connectedOrgId);
+
+	if (!subscription) {
+		return null;
+	}
+
+	return await RepoSyncState.findRepositoriesBySubscriptionIdAndRepoName(connectedOrgId, page, limit, repoName);
+};
+
+const getAllRepos = async (jiraHost: string, page: number, limit: number, repoName?: string): Promise<RepoSyncState[] | null> => {
+	const subscriptions = await Subscription.getAllForHost(jiraHost);
+
+	if (!subscriptions.length) {
+		return null;
+	}
+
+	const reposArray: RepoSyncState[][] = await Promise.all(
+		subscriptions.map(async (subscription) => {
+			const subscriptionRepos = await RepoSyncState.findRepositoriesBySubscriptionIdAndRepoName(
+				subscription.id,
+				page,
```

Comment:
Will the `page` / `limit` can work nicely with the `subscriptions.map`? I suspect they will conflicts with each other.

Reply:
I suspect you might end up needs to do something like 
```sql
select repoName, repoId from RepoSyncState rss 
where exists ( 
  select sub.id from "Subscirptions" sub 
      where sub.id = rss.subscriptionId 
           && sub.jiraHost = :jiraHost 
           && sub.orgName like '%orgName' 
) 
order by XXX asc limit :limit offset :offset
```